keras_cv/models/segmentation/deeplabv3plus.py

```diff
@@ -0,0 +1,198 @@
+            from the data should be represented by integers with range
+            [0, classes).
+        backbone: a backbone for the model, expected to be a KerasCV model.
+            Typically ResNet50V2 or ResNet101V2. Default 'low_level_feature_layer' assumes
+            either and uses 'v2_stack_1_block4_1_relu' for them by default.
+        layer_names: the layer names for the low-level and high-level features to use for encoding/decoding
+            spatial information for the supplied backbone, respectively.
+        weights: weights for the complete DeepLabV3Plus model. one of `None` (random
+            initialization), a pretrained weight file path, or a reference to
+            pre-trained weights (e.g. 'imagenet/classification') (see available
+            pre-trained weights in weights.py)
+        spatial_pyramid_pooling: also known as Atrous Spatial Pyramid Pooling (ASPP).
+            Performs spatial pooling on different spatial levels in the pyramid, with
+            dilation.
+        segmentation_head: an optional `tf.keras.Layer` that predict the segmentation
+            mask based on feature from backbone and feature from decoder.
+        segmentation_head_activation: default 'softmax', the activation layer to apply after
+            the segmentation head. Should be synchronized with the backbone's final activation
+    """
+
+    def __init__(
+        self,
+        classes,
+        backbone,
+        spatial_pyramid_pooling=None,
+        segmentation_head=None,
+        segmentation_head_activation="softmax",
+        input_shape=(None, None, 3),
+        input_tensor=None,
+        layer_names=(None, None),
+        weights=None,
+        **kwargs,
+    ):
+
+        if not isinstance(backbone, tf.keras.layers.Layer):
+            raise ValueError(
+                "Backbone need to be a `tf.keras.layers.Layer`, " f"received {backbone}"
+            )
+
+        if weights and not tf.io.gfile.exists(
+            parse_weights(weights, True, "deeplabv3plus")
+        ):
+            raise ValueError(
+                "The `weights` argument should be either `None` or the path to the "
+                f"weights file to be loaded. Weights file not found at location: {weights}"
+            )
+
+        inputs = utils.parse_model_inputs(input_shape, input_tensor)
+        x = inputs
+
+        if input_shape[0] is None and input_shape[1] is None:
+            input_shape = backbone.input_shape[1:]
+
+        if input_shape[0] is None and input_shape[1] is None:
+            raise ValueError(
+                "Input shapes for both the backbone and DeepLabV3Plus are `None`."
+            )
+
+        if layer_names == (None, None):
+            if "res" in backbone.name:
+                low_level_output = backbone.get_layer("v2_stack_0_block3_out").output
+                high_level_output = backbone.get_layer("v2_stack_3_block3_out").output
```

Comment:
This is a little too hacky, can you try to use either `ResNet50.as_backbone` or `ResNet50V2.as_backbone` for now, and add a TODO to update `as_backbone` to return customized list of levels

Reply:
Let me know when this is ready for another (probably final) round of review, thanks!

Reply:
Sure thing. With the change, the param count goes from 11M to 39M and I'm not sure what the mIOU would be. Is that okay while we're waiting for the custom level list of `as_backbone()`?